src/wallet/wallet_list.c

```diff
@@ -139,22 +139,29 @@ uint8_t get_valid_wallet_meta_data_list(wallet_list_t *list) {
   return count;
 }
 
-bool get_wallet_data_by_id(const uint8_t *wallet_id, Wallet *wallet) {
+bool get_wallet_data_by_id(const uint8_t *wallet_id,
+                           Wallet *wallet,
+                           rejection_cb *reject_cb) {
   if ((NULL == wallet_id) || (NULL == wallet)) {
     return false;
   }
 
   for (uint8_t wallet_idx = 0; wallet_idx < MAX_WALLETS_ALLOWED; wallet_idx++) {
     wallet_state state = INVALID_WALLET;
-    if (true != wallet_is_filled(wallet_idx, &state)) {
+    if (wallet_is_filled(wallet_idx, &state)) {
       continue;
     }
 
-    if (VALID_WALLET == state) {
-      if (0 == memcmp(wallet_id, get_wallet_id(wallet_idx), WALLET_ID_SIZE)) {
-        if (true == is_wallet_locked(wallet_idx)) {
-          return false;
+    if (0 == memcmp(wallet_id, get_wallet_id(wallet_idx), WALLET_ID_SIZE)) {
+      // Check if wallet is in usable state or not
+      if (INVALID_WALLET == state || is_wallet_locked(wallet_idx) ||
+          is_wallet_partial(wallet_idx) || is_wallet_unverified(wallet_idx) ||
+          is_wallet_share_not_present(wallet_idx)) {
```

Comment:
Why not simply check 
`(VALID_WALLET != state) || is_wallet_locked(wallet_idx) || is_wallet_partial(wallet_idx)`

Reply:
Thanks for this suggestion